src/components/ChattingScreen.js

```diff
@@ -1,31 +1,136 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
 
 import Header from './Header';
 import MessageSender from './MessageSender';
 
 export default function ChattingScreen() {
-	const EUNKO = 'https://img.techpowerup.org/200908/eun.png';
-	const COOL = 'https://img.techpowerup.org/200908/NjRiY2JjOGU5YzQz.png';
-
-	const MSGLIST = [
-		{ user: true, content: '안녕하세요 12기 프론트엔드 개발자분들' },
-		{ user: true, content: '저희의 대화를 마음껏 조작해보세요 💌' },
-		{ user: true, content: '상단에 프로필을 눌러서 발신자 변경하면 됩니당~' },
-		{ user: false, content: '안녕 은아' },
-		{ user: false, content: '뭐해 ?' },
-		{ user: true, content: '시원아 넌 최고의 팀장이야' },
-		{ user: false, content: '나도 아니까  ' },
-		{ user: false, content: '그만 말해줘도 돼' },
-		{ user: true, content: 'ㅠㅠ' },
-	];
-
-	return (
-		<Wrapper>
-			<Header></Header>
-			<MessageSender></MessageSender>
-		</Wrapper>
-	);
+  const EUNKO = 'https://img.techpowerup.org/200908/eun.png';
+  const COOL = 'https://img.techpowerup.org/200908/NjRiY2JjOGU5YzQz.png';
+
+  const MSGLIST = [
+    { user: true, content: '안녕하세요 12기 프론트엔드 개발자분들' },
+    { user: true, content: '저희의 대화를 마음껏 조작해보세요 💌' },
+    { user: true, content: '상단에 프로필을 눌러서 발신자 변경하면 됩니당~' },
+    { user: false, content: '안녕 은아' },
+    { user: false, content: 'ㅁㅎ?' },
+    { user: true, content: '시원아 넌 최고의 팀장이야' },
+    { user: false, content: '나도 아니까  ' },
+    { user: false, content: '그만 말해줘도 돼' },
+    { user: true, content: 'ㅠㅠ' },
+  ];
+
+  const [currentUser, setCurrentUser] = useState(false); // 현재 채팅하는 사람
+  const [newMSGList, setNewMSGList] = useState(MSGLIST); // Re-Rendering하기 위하여 생성
+
+  useEffect(() => {
+    window.scrollBy(0, window.innerHeight);
+  });
+
+  const addMessageToList = (message) => {
+    setNewMSGList(
+      newMSGList.concat([
+        {
+          user: currentUser,
+          content: message,
+        },
+      ])
+    );
+  };
+
+  const changeUser = () => {
+    setCurrentUser(!currentUser);
+  };
+
+  const chattingList = newMSGList.map((msg, index) => {
+    return (
+      <MessageLine key={index} user={msg.user} profileImg={msg.user ? EUNKO : COOL} message={msg.content}></MessageLine>
+    );
+  });
+
+  return (
+    <Wrapper>
+      <Header
+        user={currentUser ? '고은' : '정쿨'}
+        profileImg={currentUser ? EUNKO : COOL}
+        changeUser={changeUser}
+      ></Header>
+      <div className="emptyChat"></div>
+      {chattingList}
+      <div className="emptyChat"></div>
+      <MessageSender addToList={addMessageToList}></MessageSender>
+    </Wrapper>
+  );
+}
+
+function MessageLine({ user, profileImg, message }) {
+  if (user) {
+    return (
+      <LeftUser>
+        <img src={profileImg} alt="leftUser" />
+        <ChatStyle>{message}</ChatStyle>
+      </LeftUser>
+    );
+  } else {
+    return (
+      <RightUser>
+        <img src={profileImg} alt="rightUser" />
+        <ChatStyle>{message}</ChatStyle>
+      </RightUser>
+    );
+  }
```

Comment:
이 로직도 너무 좋은데, 제 의견이지만 반복되는 컴포넌트가 많아서 이렇게도 작성할 수 있을 것 같아요 ㅎㅎ
`styled-components`로 작성한 것도 `props`를 넘겨줄 수 있답니당.
```suggestion
function MessageLine({ user, profileImg, message }) {
	return (
		<Message sending={!user}>
			<Img src={profileImg} />
			<ChatStyle>{message}</ChatStyle>
		</Message>
	);
}
```

Reply:
와 이 방법... styled-component에서 props를 사용하고 싶었는데.. div로 선언하니 안 먹더라고요... Message로 선언해주니 되네요 ㅠ 공유 감사합니다!